### PR TITLE
Tests for spatial.py, remove old 'region'

### PR DESCRIPTION
**Pull request by RondeauG**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [x] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [x] (If applicable) Tests have been added.
- [x] This PR does not seem to break the templates.
- [x] CHANGES.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

* Tests for the functions in `xs.spatial`
* Small bugfix in `creep_weights` for an edge case where the grid is very small, `n` is high, and `mode=wrap`.
* Refactored `subset` to separate into multiple smaller functions.

### Does this PR introduce a breaking change?

* For 2D grids, the grid resolution estimation will now take the maximum difference, instead of the difference of the first indexes. The impact should be minimal.
* Remove old (deprecated) instances of `region`
* Remove the deprecated `xs.extract.clisops_subset()`

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -6,17 +6,25 @@ v0.9.0 (unreleased)
...
+* Removed ``xscen.extract.clisops_subset``. (:pull:`367`).
+
 Internal changes
 ^^^^^^^^^^^^^^^^
 * Updated the `cookiecutter` template to the latest version. (:pull:`358`):
     * Addresses a handful of misconfigurations in the GitHub Workflows.
     * Added a few free `grep`-based hooks for finding unwanted artifacts in the code base.
     * Updated `ruff` to v0.2.0 and `black` to v24.2.0.
+* Refactored ``xs.spatial.subset`` into smaller functions. (:pull:`367`).
+* Added more tests. (:pull:`367).
```
Comment (RondeauG) [inline CHANGES.rst:21]:
```suggestion
* Added more tests. (:pull:`367`).
```

**Review by juliettelavoie — APPROVED** [review]
